manifests/init.pp

```diff
@@ -8,128 +8,94 @@
-  $cname           = $::certs::params::cname,
-  $generate        = $::certs::params::generate,
-  $regenerate      = $::certs::params::regenerate,
-  $regenerate_ca   = $::certs::params::regenerate_ca,
-  $deploy          = $::certs::params::deploy,
-  $ca_common_name  = $::certs::params::ca_common_name,
-  $country         = $::certs::params::country,
-  $state           = $::certs::params::state,
-  $city            = $::certs::params::city,
-  $org             = $::certs::params::org,
-  $org_unit        = $::certs::params::org_unit,
-
-  $expiration      = $::certs::params::expiration,
-  $ca_expiration   = $::certs::params::ca_expiration,
-
-  $server_cert     = $::certs::params::server_cert,
-  $server_key      = $::certs::params::server_key,
-  $server_cert_req = $::certs::params::server_cert_req,
-  $server_ca_cert  = $::certs::params::server_ca_cert,
-
-  $pki_dir         = $::certs::params::pki_dir,
-  $ssl_build_dir   = $::certs::params::ssl_build_dir,
-
-  $user            = $::certs::params::user,
-  $group           = $::certs::params::group,
-
-  $default_ca_name = $::certs::params::default_ca_name,
-  $server_ca_name  = $::certs::params::server_ca_name,
+  Stdlib::Absolutepath $log_dir = $::certs::params::log_dir,
+  String $node_fqdn = $::certs::params::node_fqdn,
+  Array[String] $cname = $::certs::params::cname,
+  Boolean $generate = $::certs::params::generate,
+  Boolean $regenerate = $::certs::params::regenerate,
+  Boolean $regenerate_ca = $::certs::params::regenerate_ca,
+  Boolean $deploy = $::certs::params::deploy,
+  String $ca_common_name = $::certs::params::ca_common_name,
+  String[2,2] $country = $::certs::params::country,
+  String $state = $::certs::params::state,
+  String $city = $::certs::params::city,
+  String $org = $::certs::params::org,
+  String $org_unit = $::certs::params::org_unit,
+  String $expiration = $::certs::params::expiration,
+  String $ca_expiration = $::certs::params::ca_expiration,
+  Optional[Stdlib::Absolutepath] $server_cert = $::certs::params::server_cert,
+  Optional[Stdlib::Absolutepath] $server_key = $::certs::params::server_key,
+  Optional[Stdlib::Absolutepath] $server_cert_req = $::certs::params::server_cert_req,
+  Optional[Stdlib::Absolutepath] $server_ca_cert = $::certs::params::server_ca_cert,
+  Stdlib::Absolutepath $pki_dir = $::certs::params::pki_dir,
+  Stdlib::Absolutepath $ssl_build_dir = $::certs::params::ssl_build_dir,
+  String $user = $::certs::params::user,
+  String $group = $::certs::params::group,
+  String $default_ca_name = $::certs::params::default_ca_name,
+  String $server_ca_name = $::certs::params::server_ca_name,
 ) inherits certs::params {
 
   if $server_cert {
-    validate_absolute_path($server_cert)
-    validate_absolute_path($server_cert_req)
-    validate_absolute_path($server_key)
-    validate_absolute_path($server_ca_cert)
     validate_file_exists($server_cert, $server_cert_req, $server_key, $server_ca_cert)
```

Comment:
Unrelated, but this is the only reason we still need puppet-common. I'm wondering if we can get rid of it somehow.